modules/exadsBidAdapter.js

```diff
@@ -0,0 +1,465 @@
+          video_output: validBidRequest.params.video_output ? validBidRequest.params.video_output : undefined,
+        }
+      }
+
+      return ({
+        'id': validBidRequest.params.impressionId,
+        'bidfloor': validBidRequest.params.bidfloor,
+        'bidfloorcur': validBidRequest.params.bidfloorcur,
+        'banner': {
+          'w': size[0],
+          'h': size[1],
+          'mimes': validBidRequest.params.mimes ? validBidRequest.params.mimes : undefined,
+          ext
+        },
+      });
+    });
+  }
+
+  const nativeMediaType = utils.deepAccess(validBidRequest, 'mediaTypes.native');
+
+  if (nativeMediaType != null) {
+    const nativeVersion = '1.2';
+
+    const native = {
+      'native': {
+        'ver': nativeVersion,
+        'plcmttype': 4,
+        'plcmtcnt': validBidRequest.params.native.plcmtcnt
+      }
+    };
+
+    native.native.assets = bidRequestData.imp = nativeMediaType.ortb.assets.map(asset => {
+      const newAsset = asset;
+      if (newAsset.img != null) {
+        newAsset.img.wmin = newAsset.img.h;
+        newAsset.img.hmin = newAsset.img.w;
+      }
+      return newAsset;
+    });
+
+    const imp = [{
+      'id': validBidRequest.params.impressionId,
+      'bidfloor': validBidRequest.params.bidfloor,
+      'bidfloorcur': validBidRequest.params.bidfloorcur,
+      'native': {
+        'request': JSON.stringify(native),
+        'ver': nativeVersion
+      },
+    }];
+
+    bidRequestData.imp = imp;
+  };
+
+  const videoMediaType = utils.deepAccess(validBidRequest, 'mediaTypes.video');
+
+  if (videoMediaType != null) {
+    const imp = [{
+      'id': validBidRequest.params.impressionId,
+      'bidfloor': validBidRequest.params.bidfloor,
+      'bidfloorcur': validBidRequest.params.bidfloorcur,
+      'video': {
+        'mimes': validBidRequest.params.stream.video.mimes,
+        'protocols': validBidRequest.params.stream.protocols,
+      },
+      'ext': validBidRequest.params.stream.ext
+    }];
+
+    bidRequestData.imp = imp;
+  }
+
+  utils.logInfo('PAYLOAD', bidRequestData, JSON.stringify(bidRequestData));
+  utils.logInfo('FINAL URL', endpointUrl);
+
+  return makeBidRequest(endpointUrl, bidRequestData);
+};
+
+function handleResRTB2Dot4(serverResponse, request) {
+  utils.logInfo('on handleResRTB_2_4 -> request:', request);
+  utils.logInfo('on handleResRTB_2_4 -> request json data:', JSON.parse(request.data));
+  utils.logInfo('on handleResRTB_2_4 -> serverResponse:', serverResponse);
+
+  let bidResponses = [];
+
+  if (serverResponse.hasOwnProperty('body') && serverResponse.body.hasOwnProperty('id')) {
+    utils.logInfo('Ad server response', serverResponse.body.id);
+    utils.logInfo('serverResponse.body.seatbid[0].bid[0]', serverResponse.body.seatbid[0].bid[0]);
+
+    const bidRq = JSON.parse(request.data);
+    const requestId = serverResponse.body.id;
+    const bidData = serverResponse.body.seatbid[0].bid[0];
```

Comment:
on native there can be more than 1 bid, so bid[0] would get data for only one bid

Reply:
Now it is able to manage multiple responses for all types of requests.